### ISSUE TITLE
Access response data in case of non 200 response

### ISSUE THREAD

**Issue by Zizaco** [kribblo/ajax-promises#1]
If I have an API that answer an non 200 response code with any data, there is no way of retrieving such data:

```js
try {
  // Responds 401 with: {"friendlyMessage": "Too many attempts", "errorType": "toomanytries"}
  let response = await ajaxPromises.postJson('http://localhost/api/login', params)
} catch (err) {
  console.log(err) // Error instance. Unable to access `error` or `errorMessage`
}
```

The `err` only contain an error object but not the response data.

Is there a way of accessing response data in case of error?

**Comment by kribblo** [comment]
Hey, before anything else, have you seen the [fetch](https://github.github.io/fetch/) which is the way forward in that it is coming natively to browers? If you have the chance, I'm now recommending people to switch. :)

That said, looks like it could easily be added as `err.responseText` or some such if you'd rather stick with this one!

**Comment by kribblo** [comment]
Added `err.request` when errors are thrown so you can access whatever you like. Released as 1.0.3.
